packages/flow/lib/shared.js

```diff
@@ -1,30 +1,34 @@
 /** @type {import('./shared.d').isIterable} */
-export const isIterable = (value) => (
+export const isIterable = (value) =>
 	value != null &&
-	(
-		typeof value[Symbol.iterator] === 'function' ||
-		typeof value[Symbol.asyncIterator] === 'function'
-	)
-)
+	(typeof value[Symbol.iterator] === 'function' ||
+		typeof value[Symbol.asyncIterator] === 'function')
 
 /** @type {GeneratorFunctionConstructor} */
-const GeneratorFunction = (function * () {}).constructor
+const GeneratorFunction = function* () {}.constructor
 
 /** @type {AsyncGeneratorFunctionConstructor} */
-const AsyncGeneratorFunction = (async function * () {}).constructor
+const AsyncGeneratorFunction = async function* () {}.constructor
+
+/** @type {import('./shared.d').isGenerator} */
+const isGenerator = (value) =>
+	value instanceof GeneratorFunction ||
+	value instanceof AsyncGeneratorFunction
 
 /** @type {import('./shared.d').getNormalizedGenerator} */
-export const getNormalizedGenerator = (fn) => (
+export const getNormalizedGenerator = (fn) =>
 	typeof fn !== 'function'
-		? async function * (value) {
-			yield await value
-		}
-	: (
-		fn instanceof GeneratorFunction ||
-		fn instanceof AsyncGeneratorFunction
-	)
+		? async function* (value) {
+				yield await value
+		  }
+		: isGenerator(fn)
 		? fn
-	: async function * (value) {
-		yield await fn(await value)
-	}
-)
+		: async function* (...value) {
```

Comment:
I made this change to ensure the all args from the `iterate` method and the `<For/>` are passed down correctly